pegasus-spark-analyser/src/main/java/com/xiaomi/infra/pegasus/spark/analyser/ColdBackupLoader.java

```diff
@@ -49,24 +52,28 @@ public ColdBackupLoader(ColdBackupConfig config) throws PegasusSparkException {
     LOG.info("init fds default config and get the data urls");
   }
 
-  @Override
-  public ColdBackupConfig getConfig() {
-    return coldBackupConfig;
-  }
-
   @Override
   public int getPartitionCount() {
     return partitionCount;
   }
 
   @Override
-  public Map<Integer, String> getCheckpointUrls() {
-    return checkpointUrls;
+  public ColdBackupConfig getConfig() {
+    return coldBackupConfig;
   }
 
   @Override
-  public PegasusRecord restoreRecord(RocksIterator rocksIterator) {
-    return coldBackupConfig.getDataVersion().getPegasusRecord(rocksIterator);
+  public PegasusScanner getScanner(int pid) throws PegasusSparkException, RocksDBException {
```

Comment:
```suggestion
  public PegasusScanner getColdBackupScanner(int pid) throws PegasusSparkException, RocksDBException {
```

Reply:
`getScanner` is abstract method in `PegasusScanner` which need overload to offer different `Scanner`, For example, we can add `OnlineReader` and offer `OnlineScanner`: